canonical-web-and-design/maas-ui
# Add "Storage" section to KVM summary page.

## Done

- Added storage cards to KVM summary page. I've done it mostly according to [the design](https://app.zeplin.io/project/5e538f1156434564ebb73903/screen/5ec783846b27e649abd0406a), except I've made it 3 per row instead of 4 otherwise the mount paths + storage meter get really cramped near the medium breakpoint.
- Updated the Meter component to only accept a single `label` prop, instead of having a label per datum. I think how we plan to use meter labels is too varied, so it's easier to just have complete control over how the label looks from a single JSX element.
- Changed the Header component to route to /r/kvm instead of /l/kvm
- Fixed a css bug that prevented the VMs header rendering

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Click "KVM" in the header nav and check that it takes you to the react KVM list
- Check that the KVM list has a VMs/Owners column
- Click on a KVM and check that the Storage sections matches [the design](https://app.zeplin.io/project/5e538f1156434564ebb73903/screen/5ec783846b27e649abd0406a) (except 3 per row instead of 4)

## Fixes

Fixes canonical-web-and-design/MAAS-squad#1958

## Screenshot
![0 0 0 0_8400_MAAS_r_kvm_180](https://user-images.githubusercontent.com/25733845/87495930-df7a9180-c695-11ea-8015-d42b24a8e1ba.png)
